schema/encoding/jsonschema/benchmark_complex_test.go

```diff
@@ -0,0 +1,199 @@
+package jsonschema_test
+
+import (
+	"github.com/rs/rest-layer/schema"
+	"math"
+	"time"
+)
+
+var (
+	rfc3339NanoDefault = schema.Field{
+		Description: "UTC start time in RFC3339 format with Nano second support, e.g. 2006-01-02T15:04:05.999999999Z",
+		Validator: &schema.Time{
+			TimeLayouts: []string{time.RFC3339Nano},
+		},
+	}
+)
+
+// Default modifies an existing schema.Field with a default value.
+func Default(f schema.Field, v interface{}) schema.Field {
+	f.Default = v
+	return f
+}
+
+// Required modifies an existing schema.Field to be required.
+func Required(s schema.Field) schema.Field {
+	s.Required = true
+	return s
+}
+
+// RFC3339Nano schema.Field template
+func RFC3339Nano() schema.Field {
+	return rfc3339NanoDefault
+}
+
+// String field
+func String(min, max int, description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Default:     "",
+		Validator: &schema.String{
+			MinLen: min,
+			MaxLen: max,
+		},
+	}
+}
+
+// Integer returns schema.Field template for int validation.
+func Integer(min, max int, description string) schema.Field {
+	return schema.Field{
+		Default:     min,
+		Description: description,
+		Validator: &schema.Integer{
+			Boundaries: &schema.Boundaries{
+				Min: float64(min),
+				Max: float64(max),
+			},
+		},
+	}
+}
+
+// Bool validator helper
+func Bool(description string) schema.Field {
+	return schema.Field{
+		Description: description,
+		Validator:   &schema.Bool{},
+	}
+}
+
+func getComplexSchema() (schema.Schema, schema.Schema) {
```

Comment:
Good argument. I will separate this in to two helper functions.

As an aside, this is test code. Getting too attached to excessively "clean" test code is among the worst mistakes I feel you can make because it it can become a burden on the developers; with test code you must always be prepared to completely throw it out, it's not your product - it's an expression of the current state of the implementation that helps you have comfort that the code is doing what you intended.

I do agree it should serve as a form of documentation though, but by conveying the intent.